### PR TITLE
CE-1934: Change user for moving main page while create new wiki process

### PR DESCRIPTION
**Pull request by lukaszkonieczny**

### CODE REVIEW

File: includes/wikia/tasks/Tasks/CreateNewWikiTask.class.php [hunk 1]
```diff
@@ -146,6 +146,9 @@ private function moveMainPage() {
 					'target' => $targetTitle->getPrefixedText(),
 				];
 				if ( $sourceTitle->getPrefixedText() !== $targetTitle->getPrefixedText() ) {
+					$saveUser = $wgUser;
+					$wgUser = \User::newFromName( 'Wikia' );
```
Comment (Grunny) [inline includes/wikia/tasks/Tasks/CreateNewWikiTask.class.php]: Can we put `'Wikia'` into a constant and use it here and in `protectKeyPages` where the same pattern is used?

**Comment by Grunny** [comment]
Actually, can you use the GlobalStateWrapper Damon made for just this purpose? It's a slightly cleaner way to handle the mess of MW globals. See an example in HAWelcome which had the same issue:
* https://github.com/Wikia/app/blob/dev/extensions/wikia/HAWelcome/HAWelcomeTask.php#L325-L327
* https://github.com/Wikia/app/blob/dev/extensions/wikia/HAWelcome/HAWelcomeTask.php#L334-L336

**Comment by Grunny** [comment]
LGTM.

**Comment by lukaszkonieczny** [comment]
Tested.

**Pull request merged**